staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_flags.go

```diff
@@ -203,12 +216,36 @@ func NewDeleteFlags(usage string) *DeleteFlags {
 	return &DeleteFlags{
 		FileNameFlags: &genericclioptions.FileNameFlags{Usage: usage, Filenames: &filenames, Kustomize: &kustomize, Recursive: &recursive},
 
-		Cascade:     &cascade,
-		GracePeriod: &gracePeriod,
+		CascadingStrategy: &cascadingStrategy,
+		GracePeriod:       &gracePeriod,
 
 		// add non-defaults
 		Force:   &force,
 		Timeout: &timeout,
 		Wait:    &wait,
 	}
 }
+
+func getCascadingStrategy(cascadingFlag string) (metav1.DeletionPropagation, error) {
```

Comment:
nit, you don't have to do it here, but `getCascadingStrategy` is not the exact intent, this is more like `parseCascadingFlag`.